MailerQ.Test/ApiClientTest.cs

```diff
@@ -0,0 +1,317 @@
+
+            // Act
+            var sut = new ApiClient(Mock.Of<ILogger<ApiClient>>(), options, httpClientFactoryMock.Object);
+
+            var result = sut.Get(new Pause());
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(JsonConvert.SerializeObject(result), listPausesResult);
+        }
+
+        [Fact]
+        public async void Delete_errors_should_finish_sucessfull()
+        {
+            // Arrange            
+            var options = CreateSettings("url", "token", "version");
+            var httpClientFactoryMock = new Mock<IHttpClientFactory>();
+            var mockHttpMessageHandler = new Mock<HttpMessageHandler>();
+            mockHttpMessageHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage
+                {
+                    StatusCode = HttpStatusCode.OK,
+                });
+
+            var client = new HttpClient(mockHttpMessageHandler.Object);
+            httpClientFactoryMock.Setup(_ => _.CreateClient(It.IsAny<string>())).Returns(client);
+
+            // Act
+            var sut = new ApiClient(Mock.Of<ILogger<ApiClient>>(), options, httpClientFactoryMock.Object);
+
+            var error = new Error
+            {
+                Domain = "hotmail.com",
+                Code = 421,
+                Tag = "CampaignFromUTest",
+                Cluster = true
+            };
+
+            var result = await sut.Delete(error);
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [Fact]
+        public async void Delete_pauses_should_finish_sucessfull()
+        {
+            // Arrange            
+            var options = CreateSettings("url", "token", "version");
+            var httpClientFactoryMock = new Mock<IHttpClientFactory>();
+            var mockHttpMessageHandler = new Mock<HttpMessageHandler>();
+            mockHttpMessageHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage
+                {
+                    StatusCode = HttpStatusCode.OK,
+                });
+
+            var client = new HttpClient(mockHttpMessageHandler.Object);
+            httpClientFactoryMock.Setup(_ => _.CreateClient(It.IsAny<string>())).Returns(client);
+
+            // Act
+            var sut = new ApiClient(Mock.Of<ILogger<ApiClient>>(), options, httpClientFactoryMock.Object);
+
+            var pause = new Pause
+            {
+                Domain = "hotmail.com",
+                Tag = "PauseTestApiClient2",
+                Cluster = true
+            };
+
+            var result = await sut.Delete(pause);
+
+            // Assert
+            Assert.True(result);
```

Comment:
Maybe you can use `.Verify` method of the mock object to verify it is really called.
For example:
`httpClientFactoryMock.Verify(x => x.CreateClient(It.IsAny<string>()), Times.Once)`